site/en/docs/extensions/mv3/favicon/index.md

```diff
@@ -0,0 +1,117 @@
+---
+layout: "layouts/doc-post.njk"
+title: "Fetching favicons"
+seoTitle: "Fetching with favicons in Chrome extensions"
+date: 2023-01-11
+# updated: 2023-02-14
+description: How to get a websites' favicon in a Chrome extension.
+---
+
+## Overview {: #overview }
+
+A [favicon][mdn-favicon] (short for "favorite icon") is a small icon that is displayed in the browser's address bar. Favicons are typically used to identify and differentiate websites.
+This article describes how to retrieve a website’s favicon in a Manifest version 3 extension.
+
+## Accessing a websites' favicon {: #how-to } 
+
+To retrieve the favicon of a website, you need to construct the following URL:
+
+```text
+chrome-extension://EXTENSION_ID/_favicon/?pageUrl=EXAMPLE_URL&size=FAV_SIZE
+```
+
+`EXTENSION_ID`
+: The ID of your extension.
+
+`EXAMPLE_URL`
+: The URL of the favicon's website.
+
+`FAV_SIZE`
+: The size of the favicon. For example: 16, 32, or 64.
+
+The following steps describe how to construct this URL in a Chrome extension:  
+
+### Step 1: Update the manifest {: #manifest }
+
+First, you must request the `"favicon"` permission in the [manifest][doc-manifest].
+
+```json/4
+{
+  "name": "Favicon API in a popup",
+  "manifest_version": 3,
+  ...
+  "permissions": ["favicon"],
+  ...
+}
+```
+
+{% Aside 'caution' %}
+
+The `"favicon"` permission [triggers a warning][doc-perms-warn] unless other similar permissions are already requested, such as `"tabs"` or host permission. 
+
+{% endAside %}
+
+In addition, when fetching favicons in [content scripts][doc-cs], the `"_favicon/*"` folder must be declared as a [web accessible resource][doc-war]. For example:
+
+```json/10-16
+{
+  "name": "Favicon API in content scripts",
+  "manifest_version": 3,
+  "content_scripts": [
+    {
+      "matches": ["<all_urls>"],
+      "js": ["content.js"]
+    }
+  ],
+  "permissions": ["favicon"],
+  "web_accessible_resources": [
+    {
+      "resources": ["_favicon/*"],
+      "matches": ["<all_urls>"],
+      "extension_ids": ["*"]
+    }
+  ]
+  ...
+}
+```
+
+### Step 2: Construct the URL {: #url }
+
+The following function uses [`runtime.getURL()`][runtime-geturl] to create a fully-qualified URL pointing to the `"/_favicon/"` folder. Then it returns a new query string with the URL and size of the favicon. Finally, the extension appends the image to the body. 
+
+```js
+function faviconURL(u) {
+  const url = new URL(chrome.runtime.getURL("/_favicon/"));
+  url.searchParams.set("pageUrl", u);
+  url.searchParams.set("size", "32");
+  return url.toString();
+}
+
+const img = document.createElement('img');
+img.src = faviconURL("https://www.google.com") 
+document.body.appendChild(img);
+```
+
+This example is a `www.google.com` 32px favicon URL that includes a random extension ID:
+
+```text
+chrome-extension://eghkbfdcoeikaepkldh/_favicon/?pageUrl=https%3A%2F%2Fwww.google.com&size=32
```

Comment:
Extension IDs are always 32 characters, so I don't think this one is possible. Did you want to pick one from loading the demo extension or just make this longer?

Reply:
```suggestion
chrome-extension://eghkbfdcoeikaepkldhfgphlaiojonpc/_favicon/?pageUrl=https%3A%2F%2Fwww.google.com&size=32
```
I had shortened it to avoid scrolling so much 😅. But I think you're right; we probably should stick with being accurate 👍 . Changed it!